src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -148,7 +148,6 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
                                 {
                                     break;
                                 }
-                                Thread.Sleep(1);
```

Comment:
Why is it ok to remove this?  There's no indication for why it was here, but presumably at least at some point it served a purpose.

Reply:
We are waiting for the OS to finish initializing the process or loading/unloading modules. According to my experiments (see PR description), one retry is usually enough.

Reply:
We are waiting for the OS to finish initializing the process or loading/unloading modules. According to my experiments (see PR description), one retry is usually enough.

https://github.com/dotnet/corefx/pull/21746#discussion_r125149008 /cc @JeremyKuhne

Reply:
> According to my experiments

My concern is this code has been baked like this for a long time.  Without knowing what led to its introduction, it's difficult to say whether the experiments you ran were sufficient.  Was it under similar load?  Was it on the same OS version?  Etc.

It's possible the sleep was superfluous.  But I'm not convinced.